rfc-176-replacing-static.md

```diff
@@ -0,0 +1,127 @@
+
+![Proposed asset deploy pathway](rfc-176/proposed_govuk_asset_pathway.png)
+
+### Layouts (move to `govuk_web_layout`)
+All of the layouts in `static` are banner-aware configuration wrappers around a call to the public layout component included in the [govuk_publishing_components] gem. In practise this means that applications are having to make network calls to ask another application to return them a layout that they already have access to, and could easily configure themselves if necessary.
+
+We propose that to start with we replicate this layout configuration in the gem (so that there is little additional configuration needed in the host application), and at a later stage we audit the layouts to determine if the additional configuration is something that can be provided in a default load-time step and explicitly overridden by the applications. This should give us more flexibility in terms of personalisation.
+
+### Header and Footer Content (move to `govuk_web_layout`)
+Updating header and footer content is a developer task in `static` - there are no content items that can be edited in the publishing apps. Ideally we’d prefer this not to be the case, but that’s out of the scope of this current document.
+
+We propose for the moment moving the header and footer content into the gem, where it can be included as part of the layouts also provided by the gem. This gives us a similar developer experience to today (in that editing the content is in one place), while allowing us to remove that functionality from [static].
+
+### Additional Considerations
+
+Adding a couple more gems to every frontend app may mean that frontend development may become more awkward in situations where local versions of multiple gems are needed. These are likely to be few, but we should be prepared to monitor the developer experience in these situations. Hopefully, though, the clear division of purpose between these gems and the components gem should mean that most work can be done with at most one local gem.
+
+## Proposed Roadmap
+
+There are two initial streams, reflecting the fact that the gems could potentially be worked on independently. The work in each stream is arranged so that even if we don’t take the ultimate step, we’ll still be doing useful work. After these streams are complete, a final stream of work can retire `static`.
+
+### Stream 1: Banner Gem
+- Create prototype banner gem.
+- Moving the recruitment banners into that gem, trial including them in one of the four apps which currently contain recruitment banner code.
+- If trial looks promising, move the other three apps to use the gem.
+- Move the emergency banner code into the gem, and including the gem in `static`
+- Move current global banner code code into the gem, and use that code in `static`
+- Convert the global banner code to use the same Redis system that the emergency banner uses.
+
+### Stream 2: Layout/Shared Asset Gem
```

Comment:
This work doesn't actually seem necessary for the retirement of static and could occur afterwards if I understand correctly?

If I understand right only the shared asset stuff would be impactful where if we did this before retiring static we'd have static assets duplicated by each app. However that does seem only a minor concern given we're duplicating all the other assets and on a typical page static only seems to have around ~10 assets loaded that is 1/4 of the overall page weight.

Reply:
Going to mark this as resolved as we're now not going to do a separate layout gem. We will move the layouts into the apps themselves.